web/static/js/components/remote_retro.jsx

```diff
@@ -26,7 +29,7 @@ export class RemoteRetro extends Component {
 
     if (prevProps.presences.length) {
       const prevCurrentUser = prevProps.currentUser
-      if (isNewFacilitator(prevCurrentUser, currentUser)) {
+      if (isNewFacilitator(prevCurrentUser, currentUser) && stage !== CLOSED) {
```

Comment:
hmmm... ideally we'd have unit tests for this condition, but the boilerplate necessary for triggering a `componentDidUpdate` that references props is lengthy and troublesome in and of itself. that said, it makes me think we should extract the facilitator transfer logic out of the component and into redux in the future. as it is all reliant on the `SYNC_PRESENCE_DIFF` action. The alert reducer could feasibly handle that, but it will need access to the all of the presences at that time, which it doesn't due to it's location in the state tree. i think we could solve that with redux-thunk, as that would allow us to augment the `SYNC_PRESENCE_DIFF` event with the current state of the presences.

That said, not a blocker. I'll look into that this weekend.

Reply:
Yeah, I was thinking about the same thing but decided that this was ok for now since the feature was important. I'll read a bit about redux-thunk and share my findings. 

🤘

Reply:
hey! peeped the `retroChannel` already has access to the `store`, and therefore, we could outfit the `PRESENCE_SYNC_DIFF` action with the `presences`, and have the alert reducer just handle that action. woot!